docs/design/key_classes.md

```diff
@@ -0,0 +1,55 @@
+# Key classes
+
+## `DetectedObject`
+
+This class represents a detected object received from the host agent (HA) or a remote agent (RA).
+
+### Member variables
+
+* `timestamp`: global time associated with the `state`
+* `state`: state vector (CTRV model)
```

Comment:
Should a pedestrian also be modeled using the CTRV model? Is something like CV maybe more applicable there?

Reply:
We can refactor `DetectedObject` to be a template `struct` with parameters for a state type and covariance type.